onboard/catkin_ws/src/controls/scripts/desired_state.py

```diff
@@ -74,6 +80,28 @@ def _on_power_received(self, power):
         """
         self.power = utils.parse_twist(power)
 
+    def _power_state_safety(self, power):
+        # Compares power with controller limits
+        return_status = True
+        for axis in utils.get_axes():
+            if abs(power[axis]) > self.MAX_POWER[axis]:
+                rospy.logerr(
+                    "===> Desired power exceeds maximum power in ", axis, "! Halting robot. <===")
+                self.pid_manager.soft_estop()
+                return_status = False
+        return return_status
+
+    def _twist_state_safety(self, twist):
+        # Compares twist with controller limits
+        return_status = True
+        for axis in utils.get_axes():
+            if abs(twist[axis]) > self.MAX_TWIST[axis]:
+                rospy.logerr(
+                    "===> Desired twist exceeds maximum twist in ", axis, "! Halting robot <===")
```

Comment:
Does this syntax work for logging? I think we would have to use a format string instead:
```suggestion
                    "===> Desired twist exceeds maximum twist in %s! Halting robot <===", axis)
``` 
or 
```suggestion
                    f"===> Desired twist exceeds maximum twist in {axis}! Halting robot <===")
``` 
Also can we change the first sentence of the error message to "Desired twist exceeds maximum <axis> value!"